packages/devtools/cli/src/commands/app/open.ts

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright 2022 DXOS.org
+//
+
+import { Args, Flags, ux } from '@oclif/core';
+import chalk from 'chalk';
+import { chromium } from 'playwright';
+
+import { Client, Invitation, InvitationEncoder } from '@dxos/client';
+import { range } from '@dxos/util';
+
+import { BaseCommand } from '../../base-command';
+import { hostInvitation } from '../../util';
+
+export default class Open extends BaseCommand {
+  static override enableJsonFlag = true;
+  static override description = 'Opens app with provided url and process device invitation.';
+
+  static override args = {
+    ...super.args,
+    url: Args.string({
+      description: 'App URL.',
+      required: true,
+    }),
+  };
+
+  static override flags = {
+    ...super.flags,
+    instances: Flags.integer({
+      description: 'Amount of test instances.',
+      default: 1,
+    }),
+    invite: Flags.boolean({
+      description: 'If `true` proceed device invitation for all instances.',
+      default: true,
+    }),
+  };
+
+  async run(): Promise<any> {
+    await this.execWithClient(async (client: Client) => {
+      if (!client.halo.identity.get()) {
+        this.log(chalk`{red Profile not initialized.}`);
+        return {};
+      }
+      const { args, flags } = await this.parse(Open);
+
+      const pages = await Promise.all(
+        range(flags.instances).map(async () => {
+          const browser = await chromium.launch({ headless: false });
```

Comment:
Try reusing the browser instance and instead creating separate `context`s and windows